Framework/Geometry/src/Instrument/ComponentInfoBankHelpers.cpp

```diff
@@ -56,9 +78,23 @@ bool isAnyBank(const ComponentInfo &compInfo, const size_t &idx) {
   return false;
 }
 
+/*
+ * Function: offsetFromAncestor. Returns the position of the component at the
+ * current index relative to the ancestor component at the ancestor index. Used
+ * by saveInstrument to get the pixel offsets relative directly to the bank,
+ * ignoring any intermediate assembly types.
+ *
+ * @param compInfo : Geometry::ComponentInfo Instrument cache containing the
+ * component info.
+ * @param ancestorIdx : size_t index of ancestor component which the offset is
+ * relative to
+ * @param currentIdx : size_t index of current component
+ * @return Eigen::vector3d offset of the component at the current index relative
+ * to the ancestor component
+ */
 Eigen::Vector3d
 offsetFromAncestor(const Mantid::Geometry::ComponentInfo &compInfo,
-                   const size_t ancestorIdx, const size_t &currentIdx) {
+                   const size_t ancestorIdx, const size_t currentIdx) {
```

Comment:
We should sanity check that ancestorIndex is alway > currentIndex. Otherwise should throw.

Reply:
done